apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -167,10 +187,25 @@ if ( tracksUserData ) {
 	initializeWithIdentity( tracksUserData );
 }
 
+// Enhance templates with their parsed blocks and processed titles. Key by their slug.
+const prepareTemplatesForPlugin = ( templatesBySlug, template ) => {
+	const content = replacePlaceholders( template.content, siteInformation );
+	const title = replacePlaceholders( template.title, siteInformation );
+
+	return {
+		...templatesBySlug,
+		[ template.slug ]: { ...template, title, content },
+	};
+};
+
 registerPlugin( 'page-templates', {
 	render: () => {
 		return (
-			<PageTemplatesPlugin templates={ templates } vertical={ vertical } segment={ segment } />
+			<PageTemplatesPlugin
+				templates={ reduce( templates, prepareTemplatesForPlugin, {} ) }
```

Comment:
Might be nice if this entire operation was encapsulated in a method and avoided the inline reduce entirely. Would be easier to understand if it was well named and well (inline) documented.